server/channels/app/report.go

```diff
@@ -30,3 +150,74 @@ func (a *App) getUserReport(filter *model.UserReportOptions) ([]*model.UserRepor
 
 	return userReports, nil
 }
+
+func (a *App) StartUsersBatchExport(rctx request.CTX, dateRange string) *model.AppError {
+	options := map[string]string{
+		"requesting_user_id": rctx.Session().UserId,
+		"date_range":         dateRange,
+	}
+
+	// Check for existing job
+	// TODO: Maybe make this a reusable function?
+	pendingJobs, err := a.Srv().Jobs.GetJobsByTypeAndStatus(rctx, model.JobTypeExportUsersToCSV, model.JobStatusPending)
+	if err != nil {
+		return err
+	}
+	for _, job := range pendingJobs {
+		if job.Data["date_range"] == dateRange && job.Data["requesting_user_id"] == rctx.Session().UserId {
```

Comment:
I suppose we could add the other user ids to the job data, but there are some nuances here we'd have to account for:
- What if they request the report right as the job is sending before it's been marked Done? Then the user wouldn't get anything.
- Depending on the length of time for the report to generate, what if a new user was created that the second user who asked for the report expected to see? The already generated chunks wouldn't contain that user.

I think we're over-optimizing here for a case that likely wouldn't have a huge affect on the system and while I do like the idea, I think we should revisit it elsewhere since we'd need to think it through a bit more.

Reply:
We know for sure this is super duper duper duper heavy on MySQL, so that's why I'm trying to avoid duplicate work. Can we at least try to come up with solution to avoid duplicate data processing?

Reply:
I think this has too many moving parts to be done right now, and it's out-of-scope for this PR.
Let's revisit this elsewhere.

Reply:
Deduplicating makes sense since a) we wouldn't have to overload the server with a bunch of redundant requests (eg. someone hits the Generate button 50 times) and b) there's no point in generating the same report more than once. I suppose there's a case where a user gets added and we'd want to regenerate, but I think with the right messaging we can let the user know that the system won't do that until the first one is done. At least in this case they would only have to wait for one report.